dbrequests/database.py

```diff
@@ -9,43 +10,78 @@
 
 
 class Database(object):
-    """A Database. Encapsulates a url and an SQLAlchemy engine with a pool of
-    connections.
-
-    The url to the database can be provided directly or via a credentials-
-    dictionary `creds` with keys:
-        - host
-        - db
-        - user
-        - password
-        - dialect (defaults to mysql)
-        - driver (defaults to pymysql)
+    """
+    Provides useful methods to send and retrieve data as DataFrame. Manages
+    opening and closing connections.
+
+    - db_url: (str|None|dict):
+        - str: a sqlalchemy url
+        - None: using a sqlalchemy url from the environment variable
+          'DATABASE_URL'
+        - dict: a dict with credentials and connect_args
+            - user
+            - password
+            - host      (defaults to 127.0.0.1)
+            - port      (defaults to 3306)
+            - db
+            - dialect   (defaults to mysql) -
+            - driver    (defaults to pymysql)
+            - ...: further fields are added to 'connect_args'
+    - sql_dir: (str|None) directory where to look for sql queries. Defaults to
+      '.'.
+    - connection_class: (Connection) a connection class may be provided in
+      cases where default behaviours need to be extended. I.e. add a new 'mode'
+      in send_data.
+    - escape_percentage: (bool) escape percentages when reading queries from a
+      file.
+    - remove_comments: (bool) remove comments when reading queries from a file.
+    - kwargs:
+        - creds: (dict) deprecated, provide a dict as db_url
+        - ...: all arguments are passed to sqlalchemy.create_engine
     """
 
-    def __init__(self, db_url=None, creds=None, sql_dir=None, connection_class=DefaultConnection,
+    def __init__(self, db_url=None, sql_dir=None,
+                 connection_class=DefaultConnection,
                  escape_percentage=False, remove_comments=False, **kwargs):
-        # If no db_url was provided, fallback to $DATABASE_URL or creds.
-        self.db_url = db_url or os.environ.get('DATABASE_URL')
-        self.sql_dir = sql_dir or os.getcwd()
-        if not self.db_url:
-            try:
-                user = creds['user']
-                password = creds['password']
-                host = creds['host']
-                db = creds['db']
-                dialect = creds.get('dialect', 'mysql')
-                driver = creds.get('driver', 'pymysql')
-                self.db_url = '{}+{}://{}:{}@{}/{}'.format(
-                    dialect, driver, user, password, host, db)
-            except:
-                raise ValueError('You must provide a db_url or proper creds.')
+        if db_url is None:
+            db_url = os.environ.get('DATABASE_URL')
+            if db_url is None:
+                db_url = kwargs.pop('creds', None)
+                if db_url:
```

Comment:
this is a minor comment, but if you want to test for None, we could follow [Pep 8](https://www.python.org/dev/peps/pep-0008/) here as well. 

> Also, beware of writing if x when you really mean if x is not None -- e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be false in a boolean context!

Reply:
will do. I was unsure which part is best practice. But `is None` and `is not None` is unambiguous. Makes sense.